allennlp/models/language_model.py

```diff
@@ -0,0 +1,206 @@
+from typing import Optional, Dict
+
+from overrides import overrides
+import torch
+import torch.nn as nn
+
+from allennlp.common import Params
+from allennlp.data import Vocabulary
+from allennlp.models.model import Model
+from allennlp.modules import TextFieldEmbedder
+from allennlp.common.checks import check_dimensions_match
+from allennlp.nn import InitializerApplicator, RegularizerApplicator
+
+from allennlp.modules.lm_rnn import LMRNN
+from allennlp.modules.adaptive import AdaptiveSoftmax
+from allennlp.training.metrics.perplexity import Perplexity
+
+@Model.register("WordLM")
```

Comment:
nit: I think all of our registered names are lowercase, so maybe `word-lm`? just for consistency